core/src/test/java/jenkins/widgets/HistoryPageFilterTest.java

```diff
@@ -324,4 +354,23 @@ public int getNumber() {
             return (int) queueId;
         }
     }
+
+    // A version of MockRun that will throw an exception if getQueueId or getNumber is called
+    private static class ExplodingMockRun extends MockRun {
+        public ExplodingMockRun(long queueId) throws IOException {
+            super(queueId);
+        }
+
+        @Override
+        public long getQueueId() {
+            Assert.fail("Should not get called");
+            return super.getQueueId();
+        }
+
+        @Override
+        public int getNumber() {
+            Assert.fail("Should not get called");
```

Comment:
Not entirely sure what this is doing, but perhaps you wanted to use `RunLoadCounter`, which I would expect a PR like this to be doing?